.github/actions/bouf/action.yaml

```diff
@@ -0,0 +1,137 @@
+name: Run bouf
+description: Generates signed OBS install files and updater files
+inputs:
+  gcpWorkloadIdentityProvider:
+    description: GCP Identity Provider Pool ID
+    required: true
+  gcpServiceAccountName:
+    description: Google service account name
+    required: true
+  version:
+    description: Version string (e.g., 30.0.0-rc1)
+    required: true
+  channel:
+    description: Update channel
+    required: false
+    default: 'stable'
+
+runs:
+  using: composite
+  steps:
+    - name: Extract Artifact
+      shell: pwsh
+      run: |
+        Expand-Archive -Path build\*.zip -DestinationPath build
+        Remove-Item build\*.zip
+
+    - name: Setup bouf
+      shell: pwsh
+      env:
+        BOUF_TAG: 'v0.6.0'
+        BOUF_HASH: '06861ae6d714b754602000bd51607b5b72e655b978b7e7497b0eb970b1613dd3'
+        BOUF_NSIS_HASH: 'bb0561a76cecaba62785642dd71052a9fe53458e45df108ca08867ee333b80bd'
+        GH_TOKEN: ${{ github.token }}
+      run: |
+        # Download bouf release
+        . $env:GITHUB_ACTION_PATH\Ensure-Location.ps1
+        . $env:GITHUB_ACTION_PATH\Invoke-External.ps1
+        Ensure-Location bouf
+        Invoke-External gh release download "$env:BOUF_TAG" -R "obsproject/bouf" -p "*windows*.zip" -p "*nsis*.zip"
+        
+        if ((Get-FileHash *windows*.zip -Algorithm SHA256).Hash -ne "$env:BOUF_HASH") {
```

Comment:
This will work, though unexpectedly - the `gh` glob pattern would potentially download multiple files, and `Get-FileHash <pattern>).Hash` would yield multiple hashes (one per line) then. The `-ne` expression in turn yields all the hashes that do not match the comparator (so not a `$true` or `$false` boolean), but the string will be interpreted as `$true` so the intended effect is the same.

Too broad star expressions in a glob are icky, but should be fine for now.

Reply:
Considering that this is one of our repos (well, mostly mine) I don't expect there to ever be more than one artifact per pattern as there's only one windows build or NSIS zip.

Reply:
Correct, which is why I'm not concerned. I personally just prefer code to be "hardened" logically to not rely on assumptions. Doesn't require to be changed though.

Reply:
I just updated it now to use the same pattern as the creation of those zips.